pytorch_lightning/accelerators/ddp_backend.py

```diff
@@ -86,12 +86,18 @@ def spawn_ddp_children(self, model):
         # use the same python interpreter and actually running
         command = [sys.executable] + command
 
-        # since this script sets the visible devices we replace the gpus flag with a number
-        num_gpus = torch.cuda.device_count()
-
-        if '--gpus' in command:
-            gpu_flag_idx = command.index('--gpus')
-            command[gpu_flag_idx + 1] = f'{num_gpus}'
+        # the visible devices tell us how many GPUs we want to use.
+        # when the trainer script was called the device has already been scoped by the time
+        # code reaches this point. so, to call the scripts, we need to leave cuda visible devices alone
+        # but forward the GPUs selected via environment variables
+        gpu_ids = os.environ.get('CUDA_VISIBLE_DEVICES', '')
```

Comment:
have you tested? I guess it is not correct as `CUDA_VISIBLE_DEVICES` does the mapping on cuda level so if you have applied `export CUDA_VISIBLE_DEVICES=5,6` then `torch.cuda.device_count()` gives only 2

Reply:
yes i ran myself...

if i do: --gpus '2,3' your method gave 2.... then the script called by ddp had --gpus 2 which is wrong.